### PR TITLE
protoc-gen-go 1.26.0

### PR DESCRIPTION
**Pull request by Loyalsoldier**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by Loyalsoldier** [comment]
The repo `https://github.com/golang/protobuf` has been deprecated and moved to `https://github.com/protocolbuffers/protobuf-go/tree/master/cmd/protoc-gen-go`.

BTW, is there a LICENSE for Go projects?

**Comment by nandahkrishna** [comment]
Is there a separate license for the `protoc-gen-go` subdirectory/subproject? The `LICENSE` file at the root of the repository looks like the 3-Clause BSD License.

**Comment by Loyalsoldier** [comment]
https://github.com/protocolbuffers/protobuf-go/blob/426f20bc78ebadf3c32cd6e7cdaba378ea155521/cmd/protoc-gen-go/main.go#L1-L3

I think this is the LICENSE statement.

**Comment by SMillerDev** [comment]
It's BSD 3 clause except:
> Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

was replaced by 
> Neither the name of Google Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

I doubt they even spotted the difference. Should probably let them know about it.

File: Formula/protoc-gen-go.rb [hunk 1]
```diff
@@ -1,11 +1,15 @@
 class ProtocGenGo < Formula
   desc "Go support for Google's protocol buffers"
-  homepage "https://github.com/golang/protobuf"
-  url "https://github.com/golang/protobuf/archive/v1.5.2.tar.gz"
-  sha256 "088cc0f3ba18fb8f9d00319568ff0af5a06d8925a6e6cb983bb837b4efb703b3"
+  homepage "https://github.com/protocolbuffers/protobuf-go"
+  url "https://github.com/protocolbuffers/protobuf-go/archive/v1.26.0.tar.gz"
```
Comment (carlocab) [inline Formula/protoc-gen-go.rb:4]: Why is there a change in source?

File: Formula/protoc-gen-go.rb [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+  sha256 "26218474bcf776ecf32d7d194c6bfaca8e7b4f0c087e5b595fd50fbb31409676"
   license "BSD-3-Clause"
-  revision 1
-  head "https://github.com/golang/protobuf.git"
+  head "https://github.com/protocolbuffers/protobuf-go.git"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
```
Comment (carlocab) [inline Formula/protoc-gen-go.rb:11]: Why do we need `:github_latest`? We should only use this when we know it's needed.

**Comment by carlocab** [comment]
> The repo https://github.com/golang/protobuf has been superseded by https://github.com/protocolbuffers/protobuf-go/tree/master/cmd/protoc-gen-go

Is this documented anywhere?

**Comment by Loyalsoldier** [comment]
> > The repo https://github.com/golang/protobuf has been superseded by https://github.com/protocolbuffers/protobuf-go/tree/master/cmd/protoc-gen-go
> 
> Is this documented anywhere?

image

File: Formula/protoc-gen-go.rb (hunk 2 above)
Comment (Loyalsoldier) [reply Formula/protoc-gen-go.rb:11]: Just copy from another formula. 😂

Comment (carlocab) [reply Formula/protoc-gen-go.rb:11]: Then if we don't know that we need it, let's not put this in.

File: Formula/protoc-gen-go.rb [hunk 3]
```diff
@@ -18,7 +22,7 @@ class ProtocGenGo < Formula
   depends_on "protobuf"
 
   def install
-    system "go", "build", *std_go_args, "-ldflags", "-s -w", "./protoc-gen-go"
+    system "go", "build", *std_go_args, "-ldflags", "-s -w", "./cmd/protoc-gen-go"
```
Comment (carlocab) [inline Formula/protoc-gen-go.rb:25]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/protoc-gen-go"
```
Let's use the updated syntax for passing `ldflags`.

Comment (Loyalsoldier) [reply Formula/protoc-gen-go.rb]: Done.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**